network: Add support for multiq qdisc

DaanDeMeyer opened the pull request.

yuwata reviewed (approved):
LGTM.

DaanDeMeyer commented:
@yuwata I turns out there's both "mq" and "multiq" qdiscs. I'm a bit stuck on the naming. `mq` exposes multiple hardware queues whereas `multiq` exposes multiple software queues on top of a single hardware queue if I understand correctly. So maybe `HardwareMultiQueueing` and `SoftwareMultiQueueing`?

yuwata commented:
And there is also "mqprio", huh...

DaanDeMeyer commented:
From the kconfig:

```
config NET_SCH_MULTIQ
	tristate "Hardware Multiqueue-aware Multi Band Queuing (MULTIQ)"
```

So maybe `[MultiBandQueueing]`?

yuwata commented:
From sch_multiq.c
> MODULE_DESCRIPTION("Multi queue to hardware queue mapping qdisc");

DaanDeMeyer commented:
> Another option:
> 
>     * `[MultiQueueing]`
> 
>     * `[PriorityMultiQueueing]`
> 
>     * `[BandMultiQueueing]`

Implemented it like this now and added support for `mq` as well.

yuwata commented:
How about the following?
- `[ClassfulMultiQueueing]` -> mq, the naming 'classful' comes from the first commit when it is introduced, https://github.com/torvalds/linux/commit/6ec1c69a8f6492fd25722f4762721921da074c12
- `[PriorityMultiQueueing]` -> mqprio
- `[MultiQueueing]` or `[BandMultiQueueing]`-> multiq

yuwata commented on test/test-network/systemd-networkd-tests.py:
```diff
@@ -4648,6 +4648,16 @@ def test_qdisc_ingress(self):
+
+        output = check_output('tc qdisc show dev testtun99')
+        print(output)
+        self.assertRegex(output, 'qdisc multiq 2: root')
```
For newer test code, please use `self.assertIn()`

yuwata commented on test/test-network/systemd-networkd-tests.py:
```diff
@@ -4648,6 +4648,15 @@ def test_qdisc_ingress(self):
+
+        output = check_output('tc qdisc show dev testtun99')
+        print(output)
+        self.assertRegex(output, 'qdisc mq 2: root')
```
Also here.

DaanDeMeyer commented:
Renamed and addressed comments, setting the green label
